proposals/2246-asynchronous-uploads.md

```diff
@@ -27,12 +27,22 @@ settings (related: [MSC701]).
 The server may optionally enforce a maximum age for unused media IDs to delete
 media IDs when the client doesn't start the upload in time, or when the upload
 was interrupted and not resumed in time. The server should include the maximum
-timestamp to start the upload in the `unused_expires_at` field in the response
-JSON. The recommended default expiration for starting the upload is 1 minute.
+timestamp to complete the upload in the `unused_expires_at` field in the
+response JSON. The recommended default expiration for completing the upload is 1
```

Comment:
I think with the pending limit per user, this can be increased quite a bit.

From the previous MSC discussion, it looks like we should try to prevent resource exhaustion on the server from many pending uploads, while still allowing a person on a mobile device to try to send a photo on poor network and come back later and complete the upload with a better one. To support the latter case, the upload timeout could be something like 24 hours or more.

Another idea is that the server can vary this value in response to the amount of "quota" used. If you currently have no pending uploads, you get a full day. If you already have 10 on the go that you haven't finished, we can give you only a minute to upload instead of denying outright. Probably too fancy, but it's an idea.